Upgrade to use openxlsx2 functions.

zac-gleisner opened the pull request:
Updated tests of generate_workbook() output to check for openxslx2 workbook. 
Updated demo workbook to openxlsx2 workbook.

zac-gleisner committed:
Upgrade to use openxlsx2 functions.
Updated tests of generate_workbook() output to check for openxslx2 workbook.
Updated demo workbook to openxlsx2 workbook.

zac-gleisner committed:
Fix syntax errors.

fingertipsy commented:
Hi @zac-gleisner, 
I have reviewed all suggested changes and they look great. I have yet to try out the package in practice. 
Small think I noticed is that, vignettes/structure.Rmd and vignettes/checklist.Rmd still reference the openxlsx package and functions.

fingertipsy commented:
Is this update meant to take us in the direction of [a11ytables2](https://github.com/matt-dray/a11ytables2)? Would be interesting to know, as a lense for the review. Sorry, if I missed the meeting where this was discussed.

fingertipsy commented:
Testing the new branch against CRAN aftables with the sample workbook I noticed two small differences in behaviour (CRAN on the left, branch on the right in the images):
- for me hyperlinks were printed as links and not behind the anchor text
![image](https://github.com/user-attachments/assets/320f7fa9-d310-497e-9215-d9703f1763e8)
- there was a difference in how the blank fields were exported to excel. The branch code pastes #NA rather than leaving fields blank
![image](https://github.com/user-attachments/assets/3b15270e-5adf-4ca2-a4ae-1ba1574d2cc1)

zac-gleisner committed:
Corrected the display of hyperlinks to show anchor text in the cell.
Added option to specify N/A values should be blank in the workbook output.
Changed references to openxlsx package to openxlsx2 package.

Olivia-Box-Power committed:
Minor updates following code review

Olivia-Box-Power committed:
update vignette to openxlsx2 function

Olivia-Box-Power committed:
Update internal documentation

Olivia-Box-Power committed:
Update information on styles functions in structure vignette

Olivia-Box-Power committed:
Reverted back change to URL

Olivia-Box-Power committed:
Rename list objects of paragraph and font styles

Olivia-Box-Power committed:
Fix issue with conversion of source to hyperlink

Olivia-Box-Power committed:
Update to use the same theme as openxlsx used, so links are the same colour
